aea/cli/test.py

```diff
@@ -423,33 +423,35 @@ def load_package(
         if aea_project_path
         else find_component_directory_from_component_id_in_registry
     )
-
+    if root_packages is None:
+        raise ValueError("Packages dir not set!")
     # check the path points to a valid AEA package
     package_type = determine_package_type_for_directory(package_dir)
-
     if package_type != PackageType.AGENT:
-        if root_packages is None:
-            raise ValueError("Packages dir not set!")
         component_type = ComponentType(package_type.value)
         configuration = load_component_configuration(
             component_type, package_dir, skip_consistency_check=skip_consistency_check
         )
         configuration.directory = package_dir
         load_aea_packages_recursively(configuration, package_path_finder, root_packages)
-    else:
-        perform_load_aea_package(
-            dir_=package_dir,
-            author=package_dir.parent.parent.name,
-            package_type_plural=PackageType(package_type).to_plural(),
-            package_name=package_dir.name,
+
+        test_package_dir = package_dir / AEA_TEST_DIRNAME
+        enforce(
+            test_package_dir.exists(),
+            f"tests directory in {package_dir} not found",
+            click.ClickException,
         )
 
-    test_package_dir = package_dir / AEA_TEST_DIRNAME
-    enforce(
-        test_package_dir.exists(),
-        f"tests directory in {package_dir} not found",
-        click.ClickException,
-    )
+    else:
+        if not (aea_project_path and aea_project_path.exists()):
+            raise click.ClickException(
+                "aea_project_path not spcified or does not exists"
+            )
+
+        agent_config = AEABuilder.try_to_load_agent_configuration_file(
+            cast(Path, aea_project_path)
+        )
+        load_aea_packages_recursively(agent_config, package_path_finder, root_packages)
```

Comment:
This will fail when running an agent test by path, ie using `aea test by-path PATH_TO_AGENT`

Reply:
@angrybayblade 
updated. test written